src/circllhist.c

```diff
@@ -1637,3 +1637,49 @@ hist_approx_inverse_quantile(const histogram_t *hist, const double *in, int in_s
   }
   return 0;
 }
+
+histogram_t *
+hist_create_approximation_from_adhoc(histogram_approx_mode_t mode,
+                                     const histogram_adhoc_bin_t *bins,
+                                     size_t nbins, double sum) {
+  histogram_t *h = hist_alloc_nbins(nbins);
+  double last_spread = 0;
+  for(size_t i=0; i<nbins; i++) {
+    double a = bins[i].lower;
+    double b = bins[i].upper;
+    if(abs(a) > abs(b)) {
+      a = bins[i].upper;
+      b = bins[i].lower;
+    }
+    // chose a midpoint with minimal error
+    double m = 0;
+    switch(mode) {
+      case HIST_APPROX_HIGH:
+        m = bins[i].upper - fabs(bins[i].upper) / 1000;
+        break;
```

Comment:
This seems to work, but left me confused for a while (since m does not have to be inside the ad-hoc bin).
More natural would have been to use a "convex combination" like so:

```m = 0.001 * bins[i].lower + 0.999 * bins[i].upper // select a point the upper end of the ad-hoc bin.```

Reply:
I like this.